Gordon360/Models/ViewModels/ApartmentApplicantViewModel.cs

```diff
@@ -1,15 +1,53 @@
-﻿namespace Gordon360.Models.ViewModels
+﻿using Gordon360.Static.Data;
+using System;
+using System.Linq;
+
+namespace Gordon360.Models.ViewModels
 {
     // The view model used to send/receive apartment applicant data to/from the frontend
     public class ApartmentApplicantViewModel
     {
         public int ApplicationID { get; set; }
-        public StudentProfileViewModel Profile { get; set; }
-        public string Username { get; set; }
-        public int? Age { get; set; }
-        public string Class { get; set; }
+        public PublicStudentProfileViewModel Profile { get; set; }
+        private string _username;
+        public string Username
+        {
+            get => Profile?.AD_Username ?? _username;
+            set => _username = value;
+        }
+        public string Class => Profile?.Class ?? null;
+        public DateTime? BirthDate { get; set; }
+        public int? Age
+        {
+            get
+            {
+                if (BirthDate.HasValue)
+                {
+                    DateTime nextSemester = new DateTime(DateTime.Today.Year, 9, 1); //The next semester is fall of the current year, since apartment applications are only in the spring
+                    return nextSemester.Year - BirthDate.Value.Year; // This age is meant to be approximate, so no need for leap-year compensation or an exact 'nextSemester' date
+                }
+                else { return null; }
+            }
+        }
         public string OffCampusProgram { get; set; }
         public bool Probation { get; set; }
         public int Points { get; set; }
+
+        public static implicit operator ApartmentApplicantViewModel(GET_AA_APPLICANTS_BY_APPID_Result applicantDBModel)
```

Comment:
Oh, so I have seen this used in other view models, but it never seemed necessary. Why is it necessary here? Is it because you have more complex definitions for those two getters and setters? Does this basically do the initialization?

Reply:
Actually, I think I understand this. See my comment further down where I saw you take advantage of this.